drivers/gles3/storage/material_storage.cpp

```
@@ -2285,6 +2285,61 @@ GLuint MaterialStorage::global_shader_uniforms_get_uniform_buffer() const {
 	return global_shader_uniforms.buffer;
 }
 
+//an alternative way of allocating global shader uniforms
+//allocates a single variable and returns its position in the global shader uniform buffer
+//intended to be used for debug drawing purposes and should not be used for anything substantial
+int32_t MaterialStorage::global_shader_uniforms_unit_variable_allocate() {
+	int32_t pos = _global_shader_uniform_allocate(1);
+	ERR_FAIL_COND_V_MSG(pos < 0, -1, "Too many instances using shader instance variables. Increase buffer size in Project Settings.");
+	global_shader_uniforms.buffer_usage[pos].elements = 1;
+	return pos;
+}
+
+void MaterialStorage::global_shader_uniforms_unit_variable_free(int32_t p_pos) {
+	if (p_pos >= 0) {
+		global_shader_uniforms.buffer_usage[p_pos].elements = 0;
+	}
+}
+
+void MaterialStorage::global_shader_uniforms_unit_variable_update(int32_t p_pos, const Variant &p_value) {
+	if (p_pos < 0) {
+		return;
+	}
```

Should error instead of silently returning, perhaps with:
```
ERR_FAIL_COND(p_pos < 0);
```